runelite-client/src/main/java/net/runelite/client/plugins/microbot/util/magic/Rs2Magic.java

```diff
@@ -29,6 +31,18 @@ public static boolean canCast(MagicAction magicSpell) {
             Rs2Tab.switchToMagicTab();
             sleep(150, 300);
         }
+
+        if (magicSpell.getName().toLowerCase().contains("enchant")){
+            if (Rs2Widget.findWidget(magicSpell.getName(), Arrays.stream(Rs2Widget.getWidget(218, 0).getStaticChildren()).collect(Collectors.toList())) == null) {
```

Comment:
Why was this custom code needed?

Reply:
Because the enchantment spells are grouped within a new UI (even outside of the MTA): 
![image](https://github.com/user-attachments/assets/58099a61-39e0-421b-a14b-339f16d91469)
![image](https://github.com/user-attachments/assets/8c12b3ce-42a6-4ca6-9b67-53720dc9eb77)